validator/sawtooth_validator/execution/py_executor.py

```diff
@@ -0,0 +1,178 @@
+            settings_view_factory=SettingsViewFactory(state_view_factory),
+            scheduler_type=scheduler_type,
+            invalid_observers=[])
+
+        component_service.set_check_connections(
+            transaction_executor.check_connections)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_RECEIPT_ADD_DATA_REQUEST,
+            tp_state_handlers.TpReceiptAddDataHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_EVENT_ADD_REQUEST,
+            tp_state_handlers.TpEventAddHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_DELETE_REQUEST,
+            tp_state_handlers.TpStateDeleteHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_GET_REQUEST,
+            tp_state_handlers.TpStateGetHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_SET_REQUEST,
+            tp_state_handlers.TpStateSetHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_REGISTER_REQUEST,
+            processor_handlers.ProcessorRegisterHandler(
+                transaction_executor.processor_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_UNREGISTER_REQUEST,
+            processor_handlers.ProcessorUnRegisterHandler(
+                transaction_executor.processor_manager),
+            component_thread_pool)
+
+        component_dispatcher.start()
+        component_service.start()
+
+        self._executor = transaction_executor
+        self._context_manager = context_manager
+
+        self._component_service = component_service
+        self._component_dispatcher = component_dispatcher
+
+        self._schedulers = {}
+
+    def shutdown(self):
+        self._component_dispatcher.stop()
+        self._component_service.stop()
+        self._executor.stop()
+        self._context_manager.stop()
+
+    def create_scheduler(self, state_root, force_persist):
+        handles = self._schedulers.keys()
+        if not handles:
+            index = 0
+        else:
+            possible_index = min(handles)
+            if possible_index <= 0:
+                index = max(handles)
+            else:
+                index = possible_index
+
+        if not state_root:
+            state_root = self._context_manager.get_first_root()
+
+        scheduler = self._executor.create_scheduler(
+            self._context_manager.get_squash_handler(),
+            state_root,
+            force_persist)
+
+        self._executor.execute(scheduler=scheduler)
```

Comment:
Why did we choose to move execute here instead of exposing it an a method on the Executor interface?

Reply:
That method would take in Rust schedulers?

Reply:
Yeah, it would look just like the current execute method. I am just wondering why you chose not to expose it.

Reply:
So I don't think the block publisher or block validator want a scheduler that isn't executing. The other possibility is that there would be `create_scheduler(state_root, force_persist)` and `execute(scheduler)` but I don't know if that serves a purpose and adds one more method to the interface. Do you see a reason to separate create_scheduler and execute?

Reply:
I do not see I reason, but I think we should understand why the execute method was there in the first place before we remove it.